packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -445,20 +454,56 @@ const DataTable = ({
     ],
   );
 
-  const dataTableFilters = useMemo(
-    () =>
-      columns
-        .filter((column) => column.enableColumnFilter !== false)
-        .map((column) => {
-          return {
-            id: column.accessorKey as string,
-            label: column.header,
-            variant: column.filterVariant ?? "text",
-            options: column.filterSelectOptions,
-          } as DataFilter;
-        }),
-    [columns],
-  );
+  const convertColumnToFilter = (column: DataTableColumn<DataTableRowData>) => {
+    if (column.enableColumnFilter === false) {
+      return;
+    }
+
+    return {
+      id: column.accessorKey as string,
+      label: column.header,
+      variant: column.filterVariant ?? "text",
+      options: column.filterSelectOptions,
+    } as DataFilter;
+  };
+
+  const isDataTableColumn = (
+    item: unknown,
+  ): item is DataTableColumn<DataTableRowData> => {
+    return (
+      !!item &&
+      typeof (item as DataTableColumn<DataTableRowData>).accessorKey !==
+        "undefined"
+    );
+  };
+
+  const isDataFilter = (item: unknown): item is DataFilter => {
+    return !!item && typeof (item as DataFilter).id !== "undefined";
+  };
+
+  const dataTableFilters = useMemo(() => {
+    const providedFilters = filtersProp || columns;
+    return providedFilters
```

Comment:
I wonder if we want a comment here saying "filters can be passed in or defaulting to the use of columns".

Also, are we not combining columns with filters? If they pass custom filters, we're overwriting the columns completely right?

How would they merge them together themselves?

I'm wondering if we want another prop like `hasAutomaticColumnFilters` (please pick a better name 🙏), so filters are dependent on those. Then again, having 2 props means we could potentially have no filters passed in, so we'd need to account for that in our prop types.